src/transformers/models/git/processing_git.py

```diff
@@ -0,0 +1,113 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Image/Text processor class for GIT
+"""
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding
+
+
+class GitProcessor(ProcessorMixin):
+    r"""
+    Constructs a GIT processor which wraps a CLIP image processor and a BERT tokenizer into a single processor.
+
+    [`GitProcessor`] offers all the functionalities of [`CLIPImageProcessor`] and [`BertTokenizerFast`]. See the
+    [`~GitProcessor.__call__`] and [`~GitProcessor.decode`] for more information.
+
+    Args:
+        feature_extractor ([`CLIPImageProcessor`]):
+            The image processor is a required input.
+        tokenizer ([`BertTokenizerFast`]):
+            The tokenizer is a required input.
+    """
+    feature_extractor_class = ("CLIPImageProcessor", "VideoMAEImageProcessor")
```

Comment:
@amyeroberts I'm encountering an issue with GitProcessor. Namely, I'd like to have this processor class support multiple image processors (in this case, CLIP and VideoMAE).

However, when loading a checkpoint from the hub like so:
```
from transformers import GitProcessor

processor = GitProcessor.from_pretrained("microsoft/git-base")
```

it always instantiates the feature extractor as "VideoMAEImageProcessor", even though the checkpoint on the hub defines a CLIPImageProcessor.

Do you know what could be the reason for this?

Reply:
Looking into it, it's because it's assumed that only `tokenizer_class` can be a tuple, and the elements in the tuple are always in the `(SlowTokenizer, FastTokenizer)` order. 

The relevant code is here: https://github.com/huggingface/transformers/blob/main/src/transformers/processing_utils.py#L218-L223

Reply:
Would you be interested in making this class more general?

Also, should we update `feature_extractor_class` to `image_processor_class`?

Reply:
If you would be available to open a PR for this, that would be greatly appreciated :)

Reply:
This was fixed by simply leveraging the AutoClasses, thanks @sgugger !